library/aes.c

```diff
@@ -562,6 +557,15 @@ int mbedtls_aes_setkey_enc( mbedtls_aes_context *ctx, const unsigned char *key,
         default : return( MBEDTLS_ERR_AES_INVALID_KEY_LENGTH );
     }
```

Comment:
Does this code need to be part of the optional parameter checks? Is certainly a usability issue if you dont pass the correct key size...

Reply:
It's a necessary check and can't be optional.  If you don't check key length you won'r have a valid value for `cox->nr`.